### ISSUE TITLE
How to fix "Download of TrustListItems from DGCG failed with status code: -1"?

### ISSUE THREAD

**Issue by LonieN** [eu-digital-green-certificates/dgca-verifier-service#63]
Hi anyone help me with this issue, please?

- I think that I've set a value of [DGC_GATEWAY_CONNECTOR_ENDPOINT] is wrong? But I don't know how to fix?
  > "DGC_GATEWAY_CONNECTOR_ENDPOINT=https://dgc-gateway.example.com"
- What is the valid DGCG endpoint to replace for "https://dgc-gateway.example.com"?
- My own DGCG for development installed & run success with test cmd like: 
  > curl -X GET http://localhost:8080/trustList -H "accept: application/json" -H "X-SSL-Client-SHA256: thumbprint" -H "X-SSL-Client-DN: C=country" 

Hope receive your response soon :)

![image](https://user-images.githubusercontent.com/30336860/131108320-3c70f06f-8af6-42b5-90b0-c6b15833733c.png)

**Comment by egobsv** [comment]
Hi LonieN,
You have to set up a reverse proxy in front of your Gateway, the proxy's job is to terminate the TLS/https requests, create the X-SSL-CLIENT_DN and  X-SSL-Client-SHA256 headers and send them to port 8080.  I use Envoy to do that. Once you have that setup you can do DGC_GATEWAY_CONNECTOR_ENDPOINT=https://some_domain.local

where some_domain.local  goes to your proxy.
Hope this helps.

**Comment by LonieN** [comment]
> Hi LonieN,
> You have to set up a reverse proxy in front of your Gateway, the proxy's job is to terminate the TLS/https requests, create the X-SSL-CLIENT_DN and X-SSL-Client-SHA256 headers and send them to port 8080. I use Envoy to do that. Once you have that setup you can do DGC_GATEWAY_CONNECTOR_ENDPOINT=https://some_domain.local
> 
> where some_domain.local goes to your proxy.
> Hope this helps.

Dear egobsv,
Thanks for your response but I don't know how to using Envoy to setup a reverse proxy in front of my local DGCG, I have no experience for this. I fond a link about Envoyproxy (https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/front_proxy#), I followed it but cannot setup success. Can you help me to setup front proxy for my Gateway?

Is my DGCG (localhost:808) after setup Envoy Front proxy success will be like:

DGC_GATEWAY_CONNECTOR_ENDPOINT=https://localhost:8443
DGC_GATEWAY_CONNECTOR_TLSTRUSTSTORE_PATH=front-proxy_tls_trust_store.p12
DGC_GATEWAY_CONNECTOR_TLSKEYSTORE_PATH=front-proxy_tls_key_store.p12

Thanks & Rgds.
LonieN

**Comment by egobsv** [comment]
You don't have to use Envoy, you can use any other solution as long as then appropriate  X-SSL-CLIENT_DN and X-SSL-Client-SHA256 headers are forward to DGC_GATEWAY.
Try pointing DGC_GATEWAY_CONNECTOR_ENDPOINT to you host IP address (or a local domain name) , https://localhost  will not work since that's telling your docker service that the Gateway is available in the same container, which is not the case.  

Good luck.

**Comment by LonieN** [comment]
> You don't have to use Envoy, you can use any other solution as long as the appropriate X-SSL-CLIENT_DN and X-SSL-Client-SHA256 headers are forwarded to the DGC_GATEWAY.
> Try pointing DGC_GATEWAY_CONNECTOR_ENDPOINT to your host IP address (or a local domain name) , https://localhost will not work since that's telling your docker service that the Gateway is available in the same container, which is not the case.
> 
> Good luck.

Hi egobsv,

I have installed my Nginx server success for https://dgc-gateway.example.com & I have created:

tls_trust_store.p12 from server-cert of nginx (nginx_cert_tls.pem)
tls_key_store.p12 from the KeyPair which I have created in DGCG?
But when I rerun: docker-compose up. A output still error like:
-----
Certificates download started
Maximum age of cache reached. Fetching new TrustList from DGCG.
Failed to Download certificates from DGC Gateway. Type: CSCA, status code: -1
CSCA TrustStore contains 0 trusted certificates.
Failed to Download certificates from DGC Gateway. Type: UPLOAD, status code: -1
Upload TrustStore contains 0 trusted certificates.
Fetching TrustList from DGCG
Download of TrustListItems failed. DGCG responded with status code: -1
DSC TrustStore contains 0 trusted certificates.
Certificates download finished
-----

I don't know what I did wrong? Is tls_key_store.p12 file is created from (cert_auth.pem & key_auth.pem)  which I have created in (https://github.com/eu-digital-green-certificates/dgc-gateway#start-local) ?

Thanks so much
